components/inputs/input-text.js

```diff
@@ -150,5 +249,32 @@ class InputText extends RtlMixin(LitElement) {
 		e.preventDefault();
 	}
 
+	_handleMouseEnter() {
+		this._hovered = true;
+	}
+
+	_handleMouseLeave() {
+		this._hovered = false;
+	}
+
+	_onSlotChange(e) {
+		const children = e.target.assignedNodes();
+		if (children.length > 0) {
+			const slotName = e.target.name;
+			const affectedSlotName = this.dir !== 'rtl' ? slotName : (slotName === 'left' ? 'right' : 'left');
+			const slotContent = children[0];
+			const slotContentWidth = slotContent.offsetWidth;
+			const diff = (this.slotPaddingWidth * documentFontSize - slotContentWidth) / 2;
+
+			slotContent.style[affectedSlotName] = `${diff}px`;
+			slotContent.addEventListener('focus', this._handleFocus);
+			slotContent.addEventListener('blur', this._handleBlur);
```

Comment:
Do you need event handlers on each of the slot children? I don't think the search input was doing this. I was kind of expecting if you just fired up to the focus & blur on the surrounding custom element, then the events from inner elements would bubble up.

Reply:
So with search input, tabbing from input to the search icon did not trigger the blur event (it would be triggered when the whole component had focus removed), but with the button icon in the slot, tabbing from input to the button icon triggers the blur event. I'll look into a way around this.